yaptide/celery/tasks.py

```diff
@@ -63,99 +61,119 @@ def run_single_simulation(self,
+    tmp_dir = get_tmp_dir()
+    logging.info("Temporary directory is: %s", tmp_dir)
 
-    # if the TEMP env variable is set we will use it to override the default
-    logging.info("2. TEMP is: %s", os.environ.get("TEMP", "not set"))
-    if os.environ.get("TEMP"):
-        tmp_dir = os.environ.get("TEMP")
-
-    # if the TMP env variable is set we will use it to override the default
-    logging.info("3. TMP is: %s", os.environ.get("TMP", "not set"))
-    if os.environ.get("TMP"):
-        tmp_dir = os.environ.get("TMP")
+    command_stdout, command_stderr = '', ''
+    simulated_primaries, requested_primaries = 0, 0
 
     # with tempfile.TemporaryDirectory(dir=tmp_dir) as tmp_dir_path:
     # use the selected temporary directory to create a temporary directory
     with (
         contextlib.nullcontext(tempfile.mkdtemp(dir=tmp_dir))
         if keep_tmp_files
         else tempfile.TemporaryDirectory(dir=tmp_dir)
-    ) as tmp_dir_path:
-        logging.debug("Task %s saves the files for simulation %s", task_id, files_dict.keys())
-        write_simulation_input_files(files_dict=files_dict, output_dir=Path(tmp_dir_path))
+    ) as tmp_work_dir:
+        
+        write_simulation_input_files(files_dict=files_dict, output_dir=Path(tmp_work_dir))
+        logging.debug("Generated input files: %s", files_dict.keys())
+
+        command_as_list = command_to_run_shieldhit(dir_path=Path(tmp_work_dir), task_id=task_id)
+        logging.info("Command to run SHIELD-HIT12A: %s", " ".join(command_as_list))
 
-        watcher_green_thread = None
         # we would like to monitor the progress of simulation
         # this is done by reading the log file and sending the updates to the backend
-        # if we have update_key and simulation_id the monitoring thread can submit the updates to backend
-        if update_key is not None and simulation_id is not None:
-
-            logging.info("Sending update for task %s, setting celery id %s", task_id, self.request.id)
-            send_task_update(simulation_id, task_id, update_key, {"celery_id": self.request.id})
-
-            path_to_monitor = Path(tmp_dir_path) / f"shieldhit_{int(task_id.split('_')[-1]):04d}.log"
+        # if we have update_key and simulation_id the monitoring task can submit the updates to backend
+        path_to_monitor = Path(tmp_work_dir) / f"shieldhit_{int(task_id.split('_')[-1]):04d}.log"
+        if update_key and simulation_id is not None:
 
             current_logging_level = logging.getLogger().getEffectiveLevel()
 
-            watcher_green_thread = eventlet.spawn(read_file,
-                                                  path_to_monitor,
-                                                  simulation_id,
-                                                  task_id,
-                                                  update_key,
-                                                  logging_level=current_logging_level)
```

Comment:
current_logging_level should be passed in` kwargs`, because there are other default arguments